controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```diff
@@ -519,19 +521,24 @@ func (r *Reconciler) checkPublishRequestStatus(ctx *context.VirtualMachinePublis
 		return "", false, nil
 	case vimtypes.TaskInfoStateRunning:
 		// CreateOVF is still in progress
-		// TODO: VCLCO-2927 Add task Progress info.
+		// TODO: Add task Progress info.
 		logger.V(5).Info("VM Publish is still in progress", "progress", task.Progress)
 		ctx.VMPublishRequest.MarkUploaded(corev1.ConditionFalse, vmopv1alpha1.UploadingReason, "Uploading item to content library.")
 		return "", false, nil
 	case vimtypes.TaskInfoStateSuccess:
 		// Publish request succeeds. Update Uploaded condition.
-		ctx.VMPublishRequest.MarkUploaded(corev1.ConditionTrue)
 		logger.Info("VM Publish succeeded", "result", task.Result)
 		itemID, err := parseItemIDFromTaskResult(task.Result)
 		if err != nil {
-			// Only log this error. We can retry this operation when checking if VMI is available.
+			// Don't return err here because the task result won't be updated, the error will persist.
+			// Ideally, this should never happen.
 			logger.Error(err, "failed to get uploaded item UUID")
+			errMsg := "Failed to get the uploaded item ID. This error is unrecoverable. A new VirtualMachinePublishRequest is needed to continue publishing."
```

Comment:
Please limit the line length to 80 characters. Since it's a static error message, please define as a constant in this file somewhere.